subprojects/core/src/main/groovy/org/gradle/api/internal/project/AbstractProject.java

```diff
@@ -774,6 +774,14 @@ public Object property(String propertyName) throws MissingPropertyException {
         return extensibleDynamicObject.getProperty(propertyName);
     }
 
+    public Object propertyOrDefault(String propertyName, Object defaultValue) {
+        try {
+            return property(propertyName);
+        } catch (MissingPropertyException ignored) {
```

Comment:
We should be delegating to `hasProperty()` here to determine if the property exists rather than relying on exceptions for control flow.